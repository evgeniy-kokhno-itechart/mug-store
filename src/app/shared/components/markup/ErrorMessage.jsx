import React from 'react';
import { PropTypes } from 'prop-types';

const ErrorMessage = ({ message, customClasses }) => (
  <span className={`text-danger ${customClasses}`}>
    {message}
  </span>
);

ErrorMessage.propTypes = {
  message: PropTypes.string.isRequired,
  customClasses: PropTypes.string,
};

ErrorMessage.defaultProps = {
  customClasses: '',
};

export default ErrorMessage;
