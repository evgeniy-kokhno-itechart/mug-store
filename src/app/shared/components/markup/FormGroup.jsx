import React from 'react';
import { PropTypes } from 'prop-types';

const FormGroup = ({
  name, label, error, groupClasses, labelClasses, children,
}) => (
  <div className={`form-group ${groupClasses}`}>
    <label htmlFor={name} className={labelClasses}>{label}</label>
    {children}
    {error && <div className="alert alert-danger">{error}</div>}
  </div>
);

FormGroup.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
  groupClasses: PropTypes.string,
  labelClasses: PropTypes.string,
  children: PropTypes.shape({
    type: PropTypes.string,
  }).isRequired,
};

FormGroup.defaultProps = {
  error: '',
  groupClasses: '',
  labelClasses: '',
};

export default FormGroup;
