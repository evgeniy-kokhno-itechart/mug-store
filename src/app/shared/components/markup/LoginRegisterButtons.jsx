import React from 'react';
import { Link } from 'react-router-dom';

const LoginRegisterButtons = () => (
  <React.Fragment>
    <Link to="/login" className="nav navbar-text mr-3 d-inline clickable">
        Login
    </Link>
    <Link to="/register" className="nav navbar-text d-inline clickable">
        Register
    </Link>
  </React.Fragment>
);

export default LoginRegisterButtons;
