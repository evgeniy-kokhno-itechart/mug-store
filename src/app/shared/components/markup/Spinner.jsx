import React from 'react';
import { PropTypes } from 'prop-types';

const Spinner = ({ spinnerClasses, wrapperClasses }) => (
  <div className={`text-center ${wrapperClasses}`}>
    <div className={`spinner-border ${spinnerClasses}`} role="status">
      <span className="sr-only">Loading...</span>
    </div>
  </div>
);

Spinner.propTypes = {
  spinnerClasses: PropTypes.string,
  wrapperClasses: PropTypes.string,
};

Spinner.defaultProps = {
  spinnerClasses: '',
  wrapperClasses: '',
};

export default Spinner;
