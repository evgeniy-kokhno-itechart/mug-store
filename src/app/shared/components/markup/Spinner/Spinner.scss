@import "Variables";

.spinner {
  text-align: center;
  color: $color-primary-dark;
  

  &__icon {
    animation-name: spin;
    animation-duration: 1.5s;
    animation-timing-function: linear;
    animation-iteration-count: infinite;
  }

  &--big {
    font-size: 5rem;
  }

  &--medium {
    font-size: 3rem;
  }

  &--small {
    font-size: 1rem;
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(1turn);
  }
}