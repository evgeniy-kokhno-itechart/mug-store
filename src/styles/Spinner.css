#bootstrap-overrides .spinner--big {
  width: 5rem;
  height: 5rem;
}

#bootstrap-overrides .spinner--medium {
  width: 3rem;
  height: 3rem;
}

#bootstrap-overrides .spinner--small {
  width: 1rem;
  height: 1rem;
}
