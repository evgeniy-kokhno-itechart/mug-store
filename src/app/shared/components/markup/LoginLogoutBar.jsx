import React from 'react';
import { PropTypes } from 'prop-types';
import NameLogoutButtons from './NameLogoutButtons/NameLogoutButtons';
import LoginRegisterButtons from './LoginRegisterButtons/LoginRegisterButtons';

const LoginLogoutBar = ({ currentUserName, logoutUser }) => (
  currentUserName
    ? <NameLogoutButtons currentUserName={currentUserName} logoutUser={logoutUser} />
    : <LoginRegisterButtons />
);

LoginLogoutBar.propTypes = {
  currentUserName: PropTypes.string,
  logoutUser: PropTypes.func.isRequired,
};

LoginLogoutBar.defaultProps = {
  currentUserName: '',
};

export default LoginLogoutBar;
