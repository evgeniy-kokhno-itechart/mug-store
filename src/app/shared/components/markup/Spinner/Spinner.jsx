import React from 'react';
import { PropTypes } from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import './Spinner.scss';

const Spinner = ({ spinnerClasses, wrapperClasses }) => (
  <div className={`spinner ${wrapperClasses}`}>
    <FontAwesomeIcon icon="sync-alt" className={`spinner__icon ${spinnerClasses}`} role="status" />
    <span className="screenreaders-only">Loading...</span>
  </div>
);

Spinner.propTypes = {
  spinnerClasses: PropTypes.string,
  wrapperClasses: PropTypes.string,
};

Spinner.defaultProps = {
  spinnerClasses: '',
  wrapperClasses: '',
};

export default Spinner;
