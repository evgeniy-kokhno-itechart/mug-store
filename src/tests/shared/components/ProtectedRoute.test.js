/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { BrowserRouter } from 'react-router-dom';
import { ProtectedRoute } from '../../../app/shared/components/markup/ProtectedRoute';
import Spinner from '../../../app/shared/components/markup/Spinner';

configure({ adapter: new Adapter() });

describe('<ProtectedRoute />', () => {
  let ProtectedRouteWrapper;

  it('renders About page with default props', () => {
    ProtectedRouteWrapper = shallow(
      <BrowserRouter>
        <ProtectedRoute path="/about" />
      </BrowserRouter>,
    );
    expect(ProtectedRouteWrapper).toMatchSnapshot();
  });

  it('renders LoginPage with location = testlocation prop', () => {
    ProtectedRouteWrapper = shallow(
      <BrowserRouter>
        <ProtectedRoute path="/about" location={{ pathname: 'testlocation' }} />
      </BrowserRouter>,
    );
    expect(ProtectedRouteWrapper).toMatchSnapshot();
  });

  it('renders properly with mock Component provided', () => {
    jest.mock('../../../app/shared/components/markup/Spinner', () => {});
    ProtectedRouteWrapper = shallow(
      <BrowserRouter>
        <ProtectedRoute path="/about" currentUserName="Test User Name" component={<Spinner />} />
      </BrowserRouter>,
    );
    expect(ProtectedRouteWrapper).toMatchSnapshot();
  });

  it('renders properly with the render function provided', () => {
    const mockRender = jest.fn();
    ProtectedRouteWrapper = shallow(
      <BrowserRouter>
        <ProtectedRoute path="/about" render={mockRender} />
      </BrowserRouter>,
    );
    expect(ProtectedRouteWrapper).toMatchSnapshot();
  });
});
