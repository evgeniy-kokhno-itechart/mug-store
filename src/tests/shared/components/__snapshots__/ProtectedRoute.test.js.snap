// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProtectedRoute /> renders About page with default props 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ProtectedRoute
    component={null}
    currentUserName=""
    location={null}
    path="/about"
    render={null}
  />
</Router>
`;

exports[`<ProtectedRoute /> renders LoginPage with location = testlocation prop 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ProtectedRoute
    component={null}
    currentUserName=""
    location={
      Object {
        "pathname": "testlocation",
      }
    }
    path="/about"
    render={null}
  />
</Router>
`;

exports[`<ProtectedRoute /> renders properly with mock Component provided 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ProtectedRoute
    component={
      <Spinner
        spinnerClasses=""
        wrapperClasses=""
      />
    }
    currentUserName="Test User Name"
    location={null}
    path="/about"
    render={null}
  />
</Router>
`;

exports[`<ProtectedRoute /> renders properly with the render function provided 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ProtectedRoute
    component={null}
    currentUserName=""
    location={null}
    path="/about"
    render={[MockFunction]}
  />
</Router>
`;
